pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -646,3 +651,9 @@ func (in *InlineDocument) DeepCopy() *InlineDocument {
 func HasMixedInstances(ng *NodeGroup) bool {
 	return ng.InstancesDistribution != nil && ng.InstancesDistribution.InstanceTypes != nil && len(ng.InstancesDistribution.InstanceTypes) != 0
 }
+
+// IsAMI returns true if the string is not empty or the name of a resolver. In that case it assumes
+// that it is an AMI id
+func IsAMI(amiFlag string) bool {
+	return !(amiFlag == "" || amiFlag == NodeImageResolverStatic || amiFlag == NodeImageResolverAuto || amiFlag == NodeImageResolverAutoSSM)
```

Comment:
While I understand that this explicitly checks all non-AMI ID values that `amiFlag` can take, I believe we could also do:
```suggestion
	return strings.HasPrefix("ami-")
```

Reply:
Yeah, that would make it simpler. I just wasn't sure all AMIs are like that. Are you sure they all follow that pattern?

Reply:
I can't find an official source for the ID naming but to my knowledge, (almost?) all AWS resource IDs have some prefix identifying the resource type (like `i-` for EC2 instances, `sg-` for security groups etc)